modules/light-clients/07-tendermint/types/store.go

```diff
@@ -59,29 +58,18 @@ func SetConsensusState(clientStore sdk.KVStore, cdc codec.BinaryCodec, consensus
 
 // GetConsensusState retrieves the consensus state from the client prefixed
 // store. An error is returned if the consensus state does not exist.
-func GetConsensusState(store sdk.KVStore, cdc codec.BinaryCodec, height exported.Height) (*ConsensusState, error) {
+func GetConsensusState(store sdk.KVStore, cdc codec.BinaryCodec, height exported.Height) (*ConsensusState, bool) {
 	bz := store.Get(host.ConsensusStateKey(height))
 	if bz == nil {
-		return nil, sdkerrors.Wrapf(
-			clienttypes.ErrConsensusStateNotFound,
-			"consensus state does not exist for height %s", height,
-		)
+		return nil, false
 	}
 
 	consensusStateI, err := clienttypes.UnmarshalConsensusState(cdc, bz)
 	if err != nil {
-		return nil, sdkerrors.Wrapf(clienttypes.ErrInvalidConsensus, "unmarshal error: %v", err)
-	}
-
-	consensusState, ok := consensusStateI.(*ConsensusState)
-	if !ok {
-		return nil, sdkerrors.Wrapf(
-			clienttypes.ErrInvalidConsensus,
-			"invalid consensus type %T, expected %T", consensusState, &ConsensusState{},
-		)
+		panic(err)
```

Comment:
When this functions returns an error, at least something descriptive was returned, and that error is not fatal to use panic.
Could you elaborate more on why panic is being used here. Is the error here fatal to crash the system?
If malformed bytes is unlikely to happen, do we have validation on cdc before calling UnmarshalConsensusState? I think this validation can make sure we do not run into this fatal error.

Reply:
The SDK gracefully handles panic during transaction execution. The error returned for the transaction usually wipes the original error message but indicates that something very unexpected occurred. In this case, for the panic to occur, there must be a bug in the existing code (we must of set malformed bytes in state)

Reply:
The panic will essentially bubble up the callstack of the `UpdateClient` handler and the [grpc recovery interceptor](https://github.com/cosmos/cosmos-sdk/blob/master/baseapp/grpcserver.go#L81) in the cosmos sdk will catch and recover 👍